posts/2023-12-28-Rust-1.75.0.md

```diff
@@ -0,0 +1,96 @@
+
+The Rust team is happy to announce a new version of Rust, 1.75.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via rustup, you can get 1.75.0 with:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.75.0](https://doc.rust-lang.org/nightly/releases.html#version-1750-2023-12-28).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.75.0 stable
+
+### `async fn` and return-position `impl Trait` in traits
+
+As [announced](https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html)
+last week, Rust 1.75 supports use of `async fn` and `-> impl Trait` in traits.
+However, this initial release comes with some limitations that are described in
+the [announcement post](https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#where-the-gaps-lie).
+
+It's expected that these limitations will be lifted in future releases.
+
+### Pointer byte offset APIs
+
+Raw pointers (`*const T` and `*mut T`) used to primarily support operations
+operating in units of `T`. For example, `<*const T>::add(1)` would add
+`size_of::<T>()` bytes to the pointer's address. In some cases, working with
+byte offsets is more convenient, and these new APIs avoid requiring callers to
+cast to `*const u8`/`*mut u8` first.
+
+- [`pointer::byte_add`](https://doc.rust-lang.org/stable/core/primitive.pointer.html#method.byte_add)
+- [`pointer::byte_offset`](https://doc.rust-lang.org/stable/core/primitive.pointer.html#method.byte_offset)
+- [`pointer::byte_offset_from`](https://doc.rust-lang.org/stable/core/primitive.pointer.html#method.byte_offset_from)
+- [`pointer::byte_sub`](https://doc.rust-lang.org/stable/core/primitive.pointer.html#method.byte_sub)
+- [`pointer::wrapping_byte_add`](https://doc.rust-lang.org/stable/core/primitive.pointer.html#method.wrapping_byte_add)
+- [`pointer::wrapping_byte_offset`](https://doc.rust-lang.org/stable/core/primitive.pointer.html#method.wrapping_byte_offset)
+- [`pointer::wrapping_byte_sub`](https://doc.rust-lang.org/stable/core/primitive.pointer.html#method.wrapping_byte_sub)
+
+### Code layout optimizations for rustc
+
+The Rust compiler continues to get faster, with this release including the
+application of
+[BOLT](https://github.com/llvm/llvm-project/blob/main/bolt/README.md) to our
+binary releases, bringing a 2% mean wall time improvements on our
+benchmarks. This tool optimizes the layout of the `librustc_driver.so` library
+containing most of the rustc code, allowing for better cache utilization.
```

Comment:
Is it worth mentioning this is icache utilization?

```suggestion
containing most of the rustc code, allowing for better instruction cache utilization.
```

Reply:
I had this in the original, but I'm not sure it's *all* icache utilization. Any data in the binary is also getting shifted/split.

But I don't have counters for that - perf currently only tracks blanket cache misses, not which kind of cache.